Source: Matz himself!
Upstream: Yes, r23563 (partial patch)
Reason: Fix a failing test:

From 889a0bd773ac4c4881c790e0060588f9d3149d69 Mon Sep 17 00:00:00 2001
From: matz <matz@b2dd03c8-39d4-4d8f-98ff-823fe69b080e>
Date: Sun, 24 May 2009 23:00:01 +0000
Subject: [PATCH] * test/ruby/test_hash.rb (TestHash::test_equal2): recursive hashes
   are handled properly now. ref: [ruby-core:23402]

* test/ruby/test_m17n.rb (TestM17N#test_sprintf_p): test fixed

git-svn-id: http://svn.ruby-lang.org/repos/ruby/trunk@23563 b2dd03c8-39d4-4d8f-98ff-823fe69b080e
---
 ChangeLog              |    7 +++++++
 test/ruby/test_hash.rb |    4 +++-
 test/ruby/test_m17n.rb |    2 +-
 3 files changed, 11 insertions(+), 2 deletions(-)

diff --git a/test/ruby/test_m17n.rb b/test/ruby/test_m17n.rb
index 4ddec03..eb9aa7c 100644
--- a/test/ruby/test_m17n.rb
+++ b/test/ruby/test_m17n.rb
@@ -770,7 +770,7 @@ class TestM17N < Test::Unit::TestCase
     #assert_strenc("\"\xC2\xA1\"", 'Windows-31J', s("%p") % s("\xc2\xa1"))
     assert_strenc("\"\xC2\xA1\"", 'UTF-8', u("%p") % u("\xc2\xa1"))
 
-    assert_strenc('"\xC2\xA1"', 'US-ASCII', "%10p" % a("\xc2\xa1"))
+    assert_strenc('"\xC2\xA1"', 'ASCII-8BIT', "%10p" % a("\xc2\xa1"))
     assert_strenc("       \"\xC2\xA1\"", 'EUC-JP', "%10p" % e("\xc2\xa1"))
     #assert_strenc("       \"\xC2\xA1\"", 'Windows-31J', "%10p" % s("\xc2\xa1"))
     assert_strenc("       \"\xC2\xA1\"", 'UTF-8', "%10p" % u("\xc2\xa1"))
-- 
1.6.3.1

